platform/macos/os_macos.mm

```diff
@@ -1147,6 +1147,8 @@ static void handle_interrupt(int sig) {
 }
 
 void OS_MacOS_NSApp::terminate() {
+	godot_cleanup_profiler();
```

Comment:
I believe the placement of this is problematic.

I just tried backporting this to 4.5 and ran it in a fairly large real-world project, and in this particular project you run into a crash here when shutting down the game:

https://github.com/godotengine/godot/blob/8105ff7ac735f83be221296a6bb9b22caa7d9253/core/profiling/profiling.cpp#L82

... because `OS_MacOS_NSApp::cleanup` is what tears down the `MainLoop` (i.e. `SceneTree`), which can involve GDScript stuff, and which is called as part of `[NSApp terminate:nil]` *after* `godot_cleanup_profiler` has been called in `OS_MacOS_NSApp::terminate`.

So I would imagine that `godot_cleanup_profiler` should be called at the bottom of `OS_MacOS_NSApp::cleanup` instead, unless I'm missing something?

(I'll make an actual issue for it shortly.)